### PR TITLE
Upgrade redis to v4.1.0

### PR DESCRIPTION
**Pull request by lara-mpt**
### Change description ###

Upgrade redis to v4.1.0

### CODE REVIEW

File: src/main/modules/session/index.ts [hunk 1]
```diff
@@ -37,16 +38,20 @@ export class SessionStorage {
   private getStore(app: Application) {
     const redisHost = config.get('session.redis.host');
     if (redisHost) {
-      const client = redis.createClient({
-        host: redisHost as string,
+      const redisClient = createClient({
+        socket: {
+          host: redisHost as string,
+          port: 6380,
```
Comment (benouaer) [inline src/main/modules/session/index.ts:44]: should we add this to config

File: src/main/modules/session/index.ts [hunk 2]
```diff
@@ -37,16 +38,20 @@ export class SessionStorage {
...
+          tls: true,
+          connectTimeout: 15000,
```
Comment (benouaer) [inline src/main/modules/session/index.ts:46]: maybe this as well could be made a config variable

File: src/main/modules/session/index.ts [hunk 3]
```diff
@@ -37,16 +38,20 @@ export class SessionStorage {
...
+        },
         password: config.get('session.redis.key') as string,
-        port: 6380,
-        tls: true,
-        connect_timeout: 15000,
       });
 
-      app.locals.redisClient = client;
-      return new RedisStore({ client });
+      redisClient.connect().then(() => {
```
Comment (benouaer) [inline src/main/modules/session/index.ts:51]: if possible, can we do this with async await

Comment (lara-mpt) [reply src/main/modules/session/index.ts:51]: I tried to do that yesterday but I don't think it's possible as it throws an error
`Top-level 'await' expressions are only allowed when the 'module' option is set to 'esnext'`

Comment (lara-mpt) [reply src/main/modules/session/index.ts:51]: or a similar error to that I can't remember exactly

**Review by benouaer — APPROVED** [review]

**Pull request merged**